### PR TITLE
New states for stopped and canceled goals

### PR DESCRIPTION
**Pull request by cdupuis**
With this PR we can mark goals as `stopped` to indicate a non-failure state that still prevents subsequent goals from becoming requested. 

Mainly this was needed to reduce confusion on the Autofix goal. Now when autofixes get applied it will stop the goal which is a non-failure state and skip the downstream goals. Also the descriptions are clearer now IMHO. No scary looking errors for applied auto fixes.

Additionally I started putting in support for cancelling goals. Which needs a bit more work but will come at a later point. 

This change is fully backwards-compatible.

### CODE REVIEW

File: lib/api-helper/goal/storeGoals.ts [hunk 1]
```diff
@@ -181,19 +181,23 @@ function constructProvenance(ctx: HandlerContext): SdmProvenance {
...
-        case "success":
+        case SdmGoalState.success:
             return goal.successDescription;
-        case "failure":
+        case SdmGoalState.failure:
             return goal.failureDescription;
-        case "skipped":
-            return "Skipped"; // you probably want to use something that describes the reason instead. but don't error.
+        case SdmGoalState.skipped:
+            return "Skipped";
```
Comment (ddgenome) [inline lib/api-helper/goal/storeGoals.ts]: Not worth holding this up, but should we also support a `goal.skippedDescription`?

File: lib/api-helper/listener/executeAutofixes.ts [hunk 2]
```diff
@@ -105,17 +106,24 @@ export function executeAutofixes(registrations: AutofixRegistration[]): ExecuteG
             if (editResult.edited) {
                 // Send back an error code, because we want to stop execution of goals after this
```
Comment (ddgenome) [inline lib/api-helper/listener/executeAutofixes.ts]: This comment seems to no longer be accurate.

File: lib/api-helper/goal/storeGoals.ts (hunk 1 above)
Comment (cdupuis) [reply lib/api-helper/goal/storeGoals.ts]: skipped description isn't really a thing as the description will be set by the `SkipDownstreamGoalsOnGoalFailure` event handler in sdm-core.

File: lib/api-helper/listener/executeAutofixes.ts (hunk 2 above)
Comment (cdupuis) [reply lib/api-helper/listener/executeAutofixes.ts]: good point. update coming.

**Pull request merged**